sklearn/cross_decomposition/_pls.py

```diff
@@ -230,22 +230,30 @@ def fit(self, X, Y):
         self : object
             Fitted model.
         """
-        check_consistent_length(X, Y)
+        check_consistent_length(X, y)
         X = self._validate_data(
             X, dtype=np.float64, copy=self.copy, ensure_min_samples=2
         )
-        Y = check_array(
-            Y, input_name="Y", dtype=np.float64, copy=self.copy, ensure_2d=False
+
+        if Y is not None:
+            warnings.warn(
+                "The use of Y as a parameter is currently being deprecated in favour"
+                " of y."
+            )
+            y = Y
```

Comment:
We also need to check before hand that the user does not specify both `y` and `Y` and raise a `ValueError` if this is the case.